### PR TITLE
chore: add database testing guide

### PR DESCRIPTION
**Pull request by sweatybridge**
## What kind of change does this PR introduce?

docs update

## What is the new behavior?

add database testing guide with `supabase test db`

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: apps/reference/docs/guides/cli/testing.mdx [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+id: testing
+title: Testing Database
```
Comment (kiwicopple) [inline apps/reference/docs/guides/cli/testing.mdx:3]:
```suggestion
title: Testing your Database
```

File: apps/reference/docs/guides/cli/testing.mdx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+description: How to test your schema changes locally in the database.
```
Comment (kiwicopple) [inline apps/reference/docs/guides/cli/testing.mdx:4]:
```suggestion
description: Test your database schema, tables, functions, and policies.
```

File: apps/reference/docs/guides/cli/testing.mdx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+---
+
+import Tabs from '@theme/Tabs'
+import TabItem from '@theme/TabItem'
+
+You can use the Supabase CLI to test your database. Make sure you have [installed the CLI](https://github.com/supabase/cli#getting-started) on your local machine and [linked](https://supabase.com/docs/reference/cli/usage#supabase-link) your Supabase project.
```
Comment (kiwicopple) [inline apps/reference/docs/guides/cli/testing.mdx:10]:
```suggestion
You can use the Supabase CLI to test your database. To get started:

- [install the CLI](https://github.com/supabase/cli#getting-started) on your local machine 
- [link](/docs/reference/cli/usage#supabase-link) your Supabase project
```

Comment (kiwicopple) [reply apps/reference/docs/guides/cli/testing.mdx:10]: @sweatybridge perhaps we can move the installation instructions to the docs? (doesn't need to be in this PR, just in the future)

File: apps/reference/docs/guides/cli/testing.mdx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+## Creating a test
+
+Create a tests folder inside the supabase folder:
+
+```bash
+mkdir -p ./supabase/tests/database
+```
+
+Create a new file with a test with the `.sql` extension.
```
Comment (kiwicopple) [inline apps/reference/docs/guides/cli/testing.mdx:20]:
```suggestion
Create a new file with the `.sql` extension which will contain the test.
```

**Review by kiwicopple — APPROVED** [review]

File: apps/reference/docs/guides/cli/testing.mdx (hunk 3 above)
Comment (sweatybridge) [reply apps/reference/docs/guides/cli/testing.mdx:10]: Sure, I will link to overview which has installation instructions. https://supabase.com/docs/guides/cli

**Pull request merged**